### PR TITLE
add feature template

### LINKED ISSUE #140: Create a template for feature documents
I love feature documents! 

In my conversation with @ryanfeeley today, I felt that a (simple) template would help make them better. E.g., a template could:

* Make it clear what sections in a feature document are mandatory
* Make them more consistent with each other
* Make it easier for people not as used to doing product work to participate and propose ideas

I'll leave it to you to decide on the actual template and format, but my strawman would likely include (not necessarily in this order):

# Outcomes

What are the outcomes you want to achieve?

# Assumptions

This is where you talk about what you assume to be true. This could include assumptions around what users are doing, errors, gaps, etc., based on anecdotes, opinions, testing, or data. 

# Problem Statement

Talk about current goals of the system and how they aren't being met. What outcomes are we not delivering? This can also include an explicit request for improvement that doesn't dictate a specific or concrete solution.

# Hypothesis

A high level hypothesis of how the feature you're going propose is going help us achieve the outcomes listed above. I recommend this be a sentence of the form:

We believe that
`<doing this/building this/creating this experience>`
for `<these people>`
will achieve `<this outcome>`.
We will know this is true when we see this
`<qualitative feedback>/<quantitative feedback>/<KPI change>`.

# Describe the Feature

This is where you get into the details of your feature and/or A/B test.

# Sub-hypotheses

Your hypothesis above may too big or high level to test and measure directly. In section, you can break down your hypothesis into sub-hypotheses that hopefully predict the higher level one. These might be rolled into your feature description. One consequence of the above structure of hypotheses is that they contain all the measures you need to prove or disprove them.

Thoughts?

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #140

### CODE REVIEW

**Comment by vladikoff** [comment]
@seanmonstar @rfk @ckarlof @shane-tomlinson @vbudhram @philbooth r?

File: features/TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+### Metrics
+
+> How are you going to measure the outcome / success?
+
+> **Please provide sample artifact graphs here.**
+
+## Hypothesis
+
+> A high level hypothesis of how the feature you're going propose is going help us achieve the outcomes listed above. I recommend this be a sentence of the form:
```
Comment (rfk) [inline features/TEMPLATE.md]: s/"you're going propose"/"you're proposing", although I think this is copy-pasta from @ckarlof's original

File: features/TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+> What are the outcomes you want to achieve? What is the success criteria?
+
+### Metrics
+
+> How are you going to measure the outcome / success?
+
+> **Please provide sample artifact graphs here.**
+
+## Hypothesis
```
Comment (rfk) [inline features/TEMPLATE.md]: I suggest that the hypothesis should inform the metrics, and therefore the order of these two items should be switched.

**Comment by rfk** [comment]
Yes! 👍 from me and while I'm sure there's plenty of scope for tweaking, I think we're better off getting something landed and tweaking as we go

File: features/TEMPLATE.md (hunk 2 above)
Comment (shane-tomlinson) [inline features/TEMPLATE.md]: I was going to suggest the exact same thing.

File: features/TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+****
+
+## Detailed design
+
+> This is the bulk of the RFC. Explain the design in enough detail for somebody familiar
+with the language to understand.
+This should get into specifics and corner-cases, and include examples of how the feature is used.
+
+
+### Unresolved questions (optional)
```
Comment (shane-tomlinson) [inline features/TEMPLATE.md]: I like to call out both unresolved questions and risks. Thinking through risks and what can derail the project both minimizes surprises and allows contingencies to be developed.

**Comment by shane-tomlinson** [comment]
Back to @vladikoff to update the order, and possibly add a risks section. Other than that, looks solid.

**Comment by philbooth** [comment]
Some other sections I've found helpful in the past have been: 

1. Constraints
2. User stories
3. Data
4. Mockups

Not saying we have to include them in this template, perhaps they're too niche in some cases. Just wanted to mention them here for consideration.

**Comment by vbudhram** [comment]
Besides what has already been mentioned, LGTM!

**Pull request merged**